global_routes.go

```diff
@@ -0,0 +1,203 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/Clever/amazon-kinesis-client-go/decode"
+)
+
+const defaultValueField = "value"
+const statTypeCounter = "counter"
+const statTypeGauge = "gauge"
+
+func globalRoutes(fields map[string]interface{}) []decode.AlertRoute {
+	routes := []decode.AlertRoute{}
+
+	// chain and append all global routes here
+	routes = append(routes, processMetricsRoutes(fields)...)
+	routes = append(routes, rsyslogRateLimitRoutes(fields)...)
+	routes = append(routes, gearmanRoutes(fields)...)
+	routes = append(routes, gearcmdPassfailRoutes(fields)...)
+	routes = append(routes, gearcmdHeartbeatRoutes(fields)...)
+	routes = append(routes, wagCircuitBreakerRoutes(fields)...)
+
+	return routes
+}
+
+// Metrics emitted by node-process-metrics and go-process-metrics libraries
+func processMetricsRoutes(fields map[string]interface{}) []decode.AlertRoute {
+	via, ok := fields["via"].(string)
+	if !ok {
+		return []decode.AlertRoute{}
+	}
+	if via != "process-metrics" {
+		return []decode.AlertRoute{}
+	}
+
+	_, ok = fields["source"].(string)
+	if !ok {
+		return []decode.AlertRoute{}
+	}
+	title, ok := fields["title"].(string)
+	if !ok {
+		return []decode.AlertRoute{}
+	}
+	statType, ok := fields["type"].(string)
```

Comment:
Updated here to handle collision of `type` field https://github.com/Clever/amazon-kinesis-client-go/pull/10 and updated glide deps in this repo 9a422b8